src/greenpak/i2c.py

```diff
@@ -133,72 +174,112 @@ def __empty_rd(self, addr: int):
 
     @override
     def write(self, addr: int, data: bytearray, silent: bool = False) -> bool:
+        raise NotImplementedError("SMBus wrapper doesn't support it")
+
+    @override
+    def read(
+        self, addr: int, byte_count: int, silent: bool = False
+    ) -> bytearray | None:
+        raise NotImplementedError("SMBus wrapper doesn't support it")
+
+    @override
+    def gp_read(self, i2c_addr: int, start: int, byte_count: int) -> bytearray | None:
+        try:
+            if 0 == byte_count:
+                # assume caller probing for GP devs on bus
+                self.__empty_rd(i2c_addr)  # Can/will throw if no dev found.
+                return bytearray()
+            else:
```

Comment:
You don't need to nesting for the else since you return in the if.

Reply:
ok